Lombiq.VueJs.Samples/Assets/Scripts/VueComponents/qr-card.vue

```diff
@@ -0,0 +1,121 @@
+<template>
+    <div class="container">
+        <div class="row justify-content-center mb-3">
+            <div class="col-md-6 col-sm">
+                <div class="card">
+                    <div class="card-body">
+                        <h5 v-if="!loading && !error">Scan code</h5>
+                        <qrcode-stream @decode="onDecode" @init="onInit" :track="paintOutline">
+                            <loading-indicator v-if="loading" />
+                        </qrcode-stream>
+                    </div>
+                </div>
+            </div>
+        </div>
+        <div class="row justify-content-center mb-3" v-if="error">
+            <div class="col-md-6 col-sm">
+                <p class="error">{{ error }}</p>
+            </div>
+        </div>
+        <div class="row justify-content-center mb-3" v-if="!error && !loading">
+            <div class="col-md-6 col-sm">
+                <business-card :card-id="cardId" :api-url="apiUrl" />
+            </div>
+        </div>
+    </div>
+</template>
+
+<script>
+import { QrcodeStream } from '../../../node_modules/vue-qrcode-reader';
+import LoadingIndicator from './loading-indicator.vue';
+import BusinessCard from './business-card.vue';
+
+export default {
+    components: {
+        QrcodeStream,
+        LoadingIndicator,
+        BusinessCard,
+    },
+    props: {
+        errorMessages: {
+            type: Object,
+            default: function (raw) {
+                const messages = {
+                    NotAllowedError: 'You need to grant camera access permission.',
+                    NotFoundError: 'No camera found.',
+                    NotSupportedError: 'Secure context required (HTTPS, localhost).',
+                    NotReadableError: 'The camera already in use.',
+                    OverconstrainedError: 'Installed cameras are not suitable.',
+                    StreamApiNotSupportedError: 'The stream API is not supported in this browser.',
+                    InsecureContextError: 'Camera access is only permitted in secure context. Use HTTPS or localhost rather than HTTP.',
+                    Unknown: 'Camera error.',
+                };
+
+                return {
+                    ...messages,
+                    ...(raw || {}),
+                };
+            },
+        },
+        apiUrl: {
+            type: String,
+            required: true,
+        },
+    },
+    data: function () {
+        return {
+            cardId: null,
+            error: '',
+            loading: false,
+        };
+    },
+    methods: {
+        onDecode(result) {
+            try {
+                const content = JSON.parse(result);
+
+                if (content.cardId) {
+                    this.cardId = content.cardId;
+                }
+            }
+            catch {
+                // Nothing to do here, just continue scanning.
+            }
+        },
```

Comment:
I expected the app to decode my QR code in this method, but my breakpoint in line 75 was never hit while holding my phone in front of the camera:

![Uploading image.png…]()

Reply:
@0liver I can't see your image:
![image](https://user-images.githubusercontent.com/15349881/232778841-5301ceb3-3a8d-4784-82ba-a80fde40d558.png)

Reply:
![2023-04-18_12h48_07](https://user-images.githubusercontent.com/164360/232783688-6ba59c58-3b4c-4b49-a3b3-3d3ea15906fa.jpg)

Reply:
I tried this both in Opera and Vivaldi, which are both Chromium-based browsers.

Reply:
Maybe it can be a problem, if you scan the code before placing a breakpoint, because the app restricts scanning the same code repeatedly.

Chrome:
![image](https://user-images.githubusercontent.com/15349881/232790556-e31646a4-7438-41e4-a9be-7608bc18b355.png)

Firefox:
![image](https://user-images.githubusercontent.com/15349881/232792334-2094ecf3-1c6c-49d3-97dd-b8864945d1e5.png)

Reply:
>  if you scan the code before placing a breakpoint, because the app restricts scanning the same code repeatedly.

That definitely helped 😉

So, I've hit the breakpoint now, but after continuing I see no feedback on my screen. Since I'm not scanning a business card but some other QR code I would expect to see an error message of the some sort, e.g. "An error occurred. The given QR code does not contain the expected information."

Did you anticipate that?

Reply:
This is why the `try...catch` block. I won't over do this. This is a really minimal implementation of a virtual business card service. All foreseen errors are handled, so I think this UX problem is not an issue.

Reply:
I'm just suggesting adding a simple error message like this when you scan a non-business-card QR code:

![image](https://user-images.githubusercontent.com/164360/233015408-86e72c49-fa35-4ae1-afbe-d7602cd1bffc.png)

```suggestion
        onDecode(result) {
            try {
                const content = JSON.parse(result);

                if (content.cardId) {
                    this.cardId = content.cardId;
                    this.error = '';
                }
            }
            catch {
                // Let the user know about incompatible QR codes, and continue scanning.
                this.error = 'This QR code does not represent a business card.';
            }
        },
```

Reply:
Ok, please set the issue back to to-do.